helm/dbp-keycloak/templates/deployment.yaml

```diff
@@ -0,0 +1,102 @@
+    spec:
+      serviceAccountName:  {{ include "common.serviceAccountName" . }}
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- . | toYaml | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: "keycloak"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.command }}
+          command:
+            {{- .Values.command | toYaml | nindent 16 }}
+          {{- end }}
+          args:
+            {{ if .Values.importRealms -}}
+            - "--import-realm"
+            {{- end }}
+            {{ range .Values.extraArgs -}}
+            - {{ . }}
+            {{ end }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          resources: {{- .Values.resources | toYaml | nindent 12 }}
+          ports:
+            - name: web
+              containerPort: 8080
+              protocol: TCP
+            - name: mgmt
+              containerPort: 8090
+              protocol: TCP
+          startupProbe: {{- omit .Values.startupProbe "enabled" | toYaml | nindent 12 }}
+          livenessProbe: {{- omit .Values.livenessProbe "enabled" | toYaml | nindent 12 }}
+          readinessProbe: {{- omit .Values.readinessProbe "enabled" | toYaml | nindent 12 }}
+          envFrom:
+            # configs
+            - configMapRef:
+                name: {{ template "common.names.fullname" . }}
+            ## extra config
+            {{- if .Values.extraConfig.create }}
+            - configMapRef:
+                name: "{{ .Values.extraConfig.name }}"
+            {{- end }}
+            ## list of external configs
+            {{- range .Values.extraConfig.external }}
+            - configMapRef:
+                name: "{{ . }}"
+            {{- end }}
+            # secrets
+            - secretRef:
+                name: "{{ .Values.auth.secretName }}"
+            {{- if .Values.extraSecret.create }}
+            ## extra secret
+            - secretRef:
+                name: "{{ .Values.extraSecret.name }}"
+            {{- end }}
+            ## list of external secrets
+            {{- range .Values.extraSecret.external }}
+            - secretRef:
+                name: "{{ . }}"
+            {{- end }}
+          volumeMounts:
+            {{- if .Values.realm.name }}
+            - mountPath: "/opt/keycloak/data/import"
+              name: realm-config
+              readOnly: true
+            {{- end }}
+            {{- with .Values.extraVolumeMounts }}
+            {{- . | toYaml | nindent 12 }}
+            {{- end }}
+      volumes:
+        {{- if .Values.realm.name }}
+        - name: realm-config
+          configMap:
+            name: {{ .Values.realm.name }}
+        {{- end }}
+        {{- with .Values.extraVolumes }}
+        {{- . | toYaml | nindent 8 }}
+        {{- end }}
```

Comment:
If this Part is needed or not shoud be evalued by an helper function and if statemands
If wie have no Releam config wo is mound and add and no extraVolumes we don't need this parts